documentation/service/technical/searching-by-location.md

```diff
@@ -0,0 +1,175 @@
+# Searching by Location
+
+## Index
+- [Introduction](#introduction)
+- [Our approach](#our-approach)
+- [How we store locations](#how-we-store-locations)
+  - [Meaning of SRID 4326](#meaning-of-srid-4326)
+  - [Why not just Geometry](#why-not-just-geometry)
+  - [Performance](#performance)
+    - [Spatial indexing](#spatial-indexing)
+    - [Reducing Location Polygon areas complexity](#reducing-location-polygon-areas-complexity)
+    - [Precomputing data when possible](#precomputing-data-when-possible)
+    - [Tweaking queries to improve performance](#tweaking-queries-to-improve-performance)
+    - [Increasing our database instance resources](#increasing-our-database-instance-resources)
+- [Vacancies location](#vacancies-location)
+- [How do we get the coordinates for the search location](#how-do-we-get-the-coordinates-for-the-search-location)
+- [Location Polygons](#location-polygons)
+  - [Purpose](#purpose)
+  - [Mappings](#mappings)
+- [Getting the geographical coordinates](#getting-the-geographical-coordinates)
+  - [Google Geocoding API costs](#google-geocoding-api-costs)
+
+
+## Introduction
+
+Jobseekers searching for vacancies within a relative distance from a location is a core feature in Teaching Vacancies.
+
+At the time this was written, around 70% of our vacancy searches include location in their filters.
+
+## Our approach
+
+Once a job search location is submitted, the service:
+
+1. Gets the area or coordinates for the searched location.
+
+2. Filters vacancies that, after being included by the rest of the search filters, are located within the provided radius distance from the given location.
+
+3. To allow ordering the search results by distance, computes the exact distance between each vacancy and the given location.
+
+## How we store locations
+
+Our PostgreSQL database instance has the PostGIS extension.
+This extension allows our database to store and operate with geospatial data.
+
+The data type in the database is `geometry` with the `geographic: true` flag and `SRID 4326`.
+
+What does this mean?
+
+We store the data in a geographic coordinate system (GCS) rather than a projected coordinate system (PCS).
+The coordinates are stored as latitude and longitude on the Earth's surface, following the WGS84 standard.
+
+Using the `geometry` type with `geographic: true` allows us to leverage a wide range of PostGIS geospatial functions, such as `ST_Distance`, `ST_Within`, `ST_Intersects`, and `ST_Buffer`.
+
+### Meaning of `SRID 4326`
+
+- SRID: Spatial Referencing System ID.
+- 4326: Represents spatial data using longitude and latitude coordinates on the Earth's surface as defined in the WGS84 standard.
+
```

Comment:
its slightly more than this - it specifies the Earth as an ellipsoid, and uses elliptical geometry which is *very* expensive computationally. It looks like the wrong choice for this application, as (a) we're in the UK, and (b) our distances should be road or time-based not distance (as mountains, rivers, awkward coastlines make the calculations completely wrong in some places

Reply:
There was some talk about possibly switching to commute distances (not a technical discussion but a product team discussion). I don't remember why it was discarded/non-prioritised back then.